src/app/nodetool/qgsnodeeditor.cpp

```diff
@@ -227,10 +228,22 @@ void QgsNodeEditor::updateNodeSelection()
 {
   mSelectedFeature->blockSignals( true );
   mSelectedFeature->deselectAllVertexes();
-  Q_FOREACH ( const QModelIndex& index, mTableWidget->selectionModel()->selectedRows() )
+  Q_FOREACH( const QModelIndex& index, mTableWidget->selectionModel()->selectedRows() )
   {
     int nodeIdx = mTableWidget->item( index.row(), 0 )->data( Qt::DisplayRole ).toInt();
     mSelectedFeature->selectVertex( nodeIdx );
   }
   mSelectedFeature->blockSignals( false );
 }
+
+void QgsNodeEditor::zoomToNode( QTableWidgetItem *item )
+{
+  double x, y;
+  x = mTableWidget->item( item->row(), 1 )->data( Qt::DisplayRole ).toDouble();
+  y = mTableWidget->item( item->row(), 2 )->data( Qt::DisplayRole ).toDouble();
+
+  QgsRectangle rect;
+  rect.scale( 2, x, y );
```

Comment:
@slarosa I can't test right now - but this looks a bit fragile. Couldn't this cause zooming to odd scales depending on the crs? A better approach might be to recenter the canvas without changing scale - thoughts?